tests/acl/test_acl_dscp_change.py

```diff
@@ -0,0 +1,654 @@
+    # This script's setup code re-uses same vnets for v4inv4 and v6inv4.
+    # There will be same vnet in multiple encap types.
+    # So remove vnets *after* removing the routes first.
+    for vnet in list(data[encap_type]['vnet_vni_map'].keys()):
+        data['duthost'].shell("redis-cli -n 4 del \"VNET|{}\"".format(vnet))
+
+    time.sleep(5)
+    for tunnel in list(tunnel_names.values()):
+        data['duthost'].shell(
+            "redis-cli -n 4 del \"VXLAN_TUNNEL|{}\"".format(tunnel))
+
+
+def setup_acl_tables(duthost, setUp_vnet):
+    intfs = str()
+    for intf in setUp_vnet["connected_interfaces"]:
+        intfs += ('"' + intf + '", ')
+    extra_vars = {
+        'acl_interfaces': intfs[:-2],
+        'acl_interfacesv6': intfs[:-2],
+       }
+    dest_path = os.path.join(TMP_DIR, ACL_TABLE_FILE)
+    duthost.host.options['variable_manager'].extra_vars.update(extra_vars)
+    duthost.file(path=dest_path, state='absent')
+    duthost.template(src=os.path.join(TEMPLATES_DIR, ACL_TEMPLATE_TABLE_FILE), dest=dest_path)
+    Logger.info("Creating ACL table")
+
+    duthost.shell("config load -y {}".format(dest_path))
+    time.sleep(5)
+    py_assert(
+        duthost.shell("redis-cli -n 6 hget 'ACL_TABLE_TABLE|OVERLAY_MARK_META_TEST' 'status'")['stdout'] == 'Active')
+    py_assert(duthost.shell(
+        "redis-cli -n 6 hget 'ACL_TABLE_TABLE|OVERLAY_MARK_META_TESTV6' 'status'")['stdout'] == 'Active')
+
+
+def del_acl_tables(duthost):
+    duthost.shell("redis-cli -n 4 del 'ACL_TABLE|OVERLAY_MARK_META_TEST' ")
+    duthost.shell("redis-cli -n 4 del 'ACL_TABLE|OVERLAY_MARK_META_TESTV6' ")
+
+
+def create_acl_rule(duthost, tableName, ruleNumber, priority, dscpAction, dstIp, qualifer=None):
+    # {   "ACL_RULE": {
+    #         "{{ tableName }}|RULE{{ruleNumber}}": {
+    #             "PRIORITY": {{priority}},
+    #             "DSCP_ACTION": {{dscpAction}},
+    #             {{qualifer}}
+    #             "DST_IP": {{dstIp}}
+
+    #         }
+    #     }
+    # }
+    if tableName == TABLE_V4:
+        qualifer = '"DST_IP":"{}"'.format(dstIp) + (',' + qualifer if qualifer is not None else "")
+    else:
+        qualifer = '"DST_IPV6":"{}"'.format(dstIp) + (',' + qualifer if qualifer is not None else "")
+    extra_vars = {
+        'tableName': tableName,
+        'ruleNumber': ruleNumber,
+        'priority': priority,
+        'dscpAction': dscpAction,
+        'qualifer': qualifer,
+       }
+
+    dest_path = os.path.join(TMP_DIR, ACL_RULES_FILE)
+    duthost.host.options['variable_manager'].extra_vars.update(extra_vars)
+    duthost.file(path=dest_path, state='absent')
+    duthost.template(src=os.path.join(TEMPLATES_DIR, ACL_TEMPLATE_RULE_FILE), dest=dest_path)
+    Logger.info("Creating ACL rule for %s with id %s, priority %s, dscp action %s, DST IP %s and qualifers %s",
+                tableName, ruleNumber, priority, dscpAction, dstIp, qualifer)
+
+    duthost.shell("config load -y {}".format(dest_path))
```

Comment:
Please check the status of the ACL rule to ensure it's active. Otherwise the following traffic test can be flaky.